tests/ui/deriving/deriving-all-codegen.stdout

```diff
@@ -598,7 +598,9 @@ impl<T: ::core::hash::Hash + Trait, U: ::core::hash::Hash> ::core::hash::Hash
     }
 }
 #[automatically_derived]
-impl<T: Trait, U> ::core::marker::StructuralPartialEq for Generic<T, U> { }
+impl<T: Trait, U> ::core::marker::StructuralPartialEq for Generic<T, U> where
+    T::A: {
```

Comment:
This is an empty bound? Seems weird...

Reply:
Yeah. We could add a special case to the deriving logic to skip empty bounds. OTOH that's not actually necessary, and I was happy to see the diff and know that my changes actually did something. ;)

Whatever you prefer.

Reply:
A special case would be good. I previously put a lot of effort into improving the derived impl code, to make it (a) concise, and (b) close to what a human would write. Skipping empty bounds would help on both counts.

Reply:
Done. Now this PR is a complete NOP for the generated output. :)